ServerCore/Pages/Events/FastestSolves.cshtml.cs

```diff
@@ -202,8 +202,10 @@ await PuzzleStateHelper.GetSparseQuery(_context, this.Event, null, null)
 
         private async Task updateSinglePlayerPuzzleStats(HashSet<int> authorPuzzleIds)
         {
-            if (!Event.ShouldShowSinglePlayerPuzzles)
+            bool isRegisteredUser = await this.IsRegisteredUser();
+            if ((!Event.ShouldShowSinglePlayerPuzzles) || (EventRole == EventRole.play && !isRegisteredUser))
```

Comment:
This should be handled by auth

Reply:
I agree, so I can add an auth redirect.  I'm still going to leave this though since the team puzzle section has it and it seems reasonable that somebody needs to be registered for the event to view this data.